### PR TITLE
[ACA-2989]Creating data-table-component PO to support all kind of columns, including custom

### PR DESCRIPTION
**Pull request by cristinaj**
…om adf testing package]. Currently the majority of the methods that we have on this class are working only on 'text' column type. The scope of this PR was to make them work on all types of columns (text, image, date, fileSize, location, json) and also custom columns that may be created on different applications or other pages.

**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/testing/src/lib/core/pages/data-table/column.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export abstract class Column {
+    columnName: string;
+    columnType: string;
+    locator: string;
+
+    constructor(columnName: string) {
+        this.columnName = columnName;
+    }
+
+    // this will work for any type of columnType: 'text', 'image', 'date', 'fileSize', 'location', and 'json'.
```
Comment (eromano) [inline lib/testing/src/lib/core/pages/data-table/column.ts:27]: this comment is not necessary

File: lib/testing/src/lib/core/pages/data-table/data-table-builder.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                case 'date': {
+                    datatable.addItem(new DateColumn(column.columnName));
+                    break;
+                }
+                case 'custom': {
+                    datatable.addItem(column);
+                    break;
+                }
+                default: datatable.addItem(new TextColumn(column.columnName));
+                // here cases for 'image', 'fileSize', 'location', and 'json' can be added when needed
```
Comment (eromano) [inline lib/testing/src/lib/core/pages/data-table/data-table-builder.ts:39]: this comment is not needed

**Comment by mauriziovitale** [comment]
@DenysVuika could you give us your though ?

**Comment by mauriziovitale** [comment]
> For me looks ok as far we refactor completely and we don't keep two pages

I agree that we should not have 2 pages but is not feasible to change all the apps in one go. So I would suggest to merge the new code, start to using it on ADW and once we are 100% that we are able to cover all the different scenario we can create a story related to the refactor and change 1 app at time. @eromano @DenysVuika @cristinaj

**Comment by DenysVuika** [comment]
Yeah, let’s add new code and remove old one in the second round. In that case we may have time to update all the apps without broken e2e pipeline?

**Pull request merged**